lib/grape/util/hash_parameter.rb

```diff
@@ -0,0 +1,20 @@
+require 'active_support/concern'
+module Grape
+  module Util
+    module HashParameter
+      extend ActiveSupport::Concern
+
+      def deem_hash_array?(hash)
+        return false unless hash.is_a?(Hash) && hash.keys.any? { |key| integer_string?(key) }
+        true
+      end
+
+      def integer_string?(str)
+        Integer(str)
+        true
+      rescue ArgumentError, TypeError
+        false
+      end
+    end
+  end
+end
```

Comment:
I am not in love with this. It brings an odd method, `integer_string` into whatever mixes it. I am not sure how to better refactor this, but at least not as a mixin maybe and just call class methods?